go/runtime/bundle/manager.go

```diff
@@ -244,6 +321,116 @@ func (m *Manager) Add(path string) error {
 	return nil
 }
 
+// AddTemporary adds a bundle from the given temporary bundle file.
+func (m *Manager) AddTemporary(tmpName string, labels map[string]string) error {
+	path, err := m.deriveTemporaryBundlePath(tmpName, labels)
+	if err != nil {
+		return err
+	}
+
+	if err = m.Add(path, labels); err != nil {
+		return err
+	}
+
+	_ = os.Remove(path) // Cleanup temporary bundle after adding.
+	return nil
+}
+
+// Remove removes bundles matching the given labels.
+func (m *Manager) Remove(labels map[string]string) error {
+	if err := validateLabels(labels); err != nil {
+		return err
+	}
+
+	m.store.RemoveManifestsWithLabels(labels)
```

Comment:
Is it ok to allow adding manifests without labels above, but not having a way to remove those via the manager?

Reply:
Yes, I don't think we want to allow removing components that are set through configuration?

Reply:
I don't think the `manager.Add` method is used without labels anywhere. Maybe `Add` should require at least one label for component?
Since if i see correctly components set through configuration don't even use `manager.Add? (are added via `registerManifest`). Then we only have components without labels (set through configuration) and labeled components (added via the bundle manager API).

Reply:
Oh I see, you're right. In this case we can simply fail if there are no labels defined in `validateLabels`. And also ensure at least the origin label is defined.